internal/runnerinstall/docker.go

```diff
@@ -0,0 +1,198 @@
+	defer func() { s.Abort() }()
+
+	cli, err := client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		return err
+	}
+	cli.NegotiateAPIVersion(ctx)
+
+	runnerImage := i.Config.RunnerImage
+	imageRef, err := reference.ParseNormalizedNamed(runnerImage)
+
+	imageList, err := cli.ImageList(ctx, types.ImageListOptions{
+		Filters: filters.NewArgs(filters.KeyValuePair{
+			Key:   "reference",
+			Value: reference.FamiliarString(imageRef),
+		}),
+	})
+	if err != nil {
+		return err
+	}
+
+	if len(imageList) == 0 {
+		s.Update("Pulling image %s", runnerImage)
+
+		resp, err := cli.ImagePull(ctx, reference.FamiliarName(imageRef), types.ImagePullOptions{})
+		if err != nil {
+			s.Update("Unable to pull waypoint image")
+			return err
+		}
+		defer resp.Close()
+
+		stdout, _, err := ui.OutputWriters()
+		if err != nil {
+			return err
+		}
+
+		var termFd uintptr
+		if f, ok := stdout.(*os.File); ok {
+			termFd = f.Fd()
+		}
+
+		err = jsonmessage.DisplayJSONMessagesStream(resp, s.TermOutput(), termFd, true, nil)
+		if err != nil {
+			return fmt.Errorf("unable to stream pull logs to the terminal: %s", err)
+		}
+
+	}
+
+	var waypointNetwork network.NetworkingConfig
+	if i.Config.Network != "" {
+		waypointNetwork = network.NetworkingConfig{
+			EndpointsConfig: map[string]*network.EndpointSettings{
+				i.Config.Network: {},
+			},
+		}
+	}
+
+	// The key thing in the container creation below is that the environment
+	// variables are set to the advertised address env vars which will
+	// allow our runner to connect.
+	cr, err := cli.ContainerCreate(ctx, &container.Config{
+		AttachStdout: true,
+		AttachStderr: true,
+		AttachStdin:  true,
+		OpenStdin:    true,
+		StdinOnce:    true,
+		User:         "root",
+		Image:        runnerImage,
+		Env:          opts.AdvertiseClient.Env(),
+		Cmd:          []string{"runner", "agent", "-id=" + opts.Id, "-cookie=" + opts.Cookie, "-vv"},
+		Labels: map[string]string{
+			"waypoint-type": "runner",
+		},
+	}, &container.HostConfig{
+		Privileged: true,
+		CapAdd:     []string{"CAP_DAC_OVERRIDE"},
+		Binds:      []string{"/var/run/docker.sock:/var/run/docker.sock"},
+		// These security options are required for the runner so that
+		// Docker daemonless image building works properly.
+		SecurityOpt: []string{
+			"seccomp=unconfined",
+			"apparmor=unconfined",
+		},
+	}, &waypointNetwork, nil, "waypoint-runner-"+opts.Id)
+	if err != nil {
+		return err
+	}
+
+	err = cli.ContainerStart(ctx, cr.ID, types.ContainerStartOptions{})
+	if err != nil {
+		return err
+	}
+
+	s.Update("Waypoint runner installed and started!")
+	s.Done()
+
+	return nil
+}
+
+func (i *DockerRunnerInstaller) InstallFlags(set *flag.Set) {
+	set.StringVar(&flag.StringVar{
+		Name:    "docker-runner-image",
+		Target:  &i.Config.RunnerImage,
+		Usage:   "The Docker image for the Waypoint runner.",
+		Default: "hashicorp/waypoint:latest",
+	})
+
+	set.StringVar(&flag.StringVar{
+		Name:   "docker-runner-network",
```

Comment:
On second thought, I'm getting this no matter what it seems so I'll debug a bit!

Reply:
The Docker network is optional, in case you want the runner to talk to a Waypoint server in the same Docker network. It's optional because the server might not be installed in the same Docker network as the runner you're installing, so it might not be necessary to install the runner to a Docker network at all.

Regarding why you're hitting "context deadline exceeded", you might need the `-server-tls-skip-verify` flag if your Waypoint server's TLS configuration is self-signed.

Reply:
Ah yes, I needed that flag! Can we add that flag to the example command in the docs/runner-manual section, since our default server configuration is self-signed?

Reply:
This part
```
```shell-session
 waypoint runner install -platform=docker -docker-runner-network=waypoint -server-addr=waypoint-server:9701 -server-cookie=abc123
 ✓ Connecting to: waypoint-server:9701
```